### PR TITLE
#164139783 List User Functionalities

### PR DESCRIPTION
**Pull request by mezlet**
#### What does this PR do?
List User Functionalities
#### Description of Task to be completed?
- write unit test
- Create a route for listing all authors
- Create a route for viewing an author by username
- validate inputs with `joi`
#### How should this be manually tested?
- ensure that the build is passing on travis 
- Test endpoint api/profiles
- Test endpoint api/profiles/username
#### What are the relevant pivotal tracker stories?
[#164139783](https://www.pivotaltracker.com/story/show/164139783)
#### Screenshots (if appropriate)
N/A
#### Questions:
N/A

### CODE REVIEW

File: server/controllers/profile-controller.js [hunk 1]
```diff
@@ -0,0 +1,71 @@
+import { successResponse, errorResponse } from '../utils/helpers';
+import db from '../models';
+
+const { User } = db;
+/**
+ * The controllers for users route
+ *
+ * @class UsersController
```
Comment (andela-pessien) [inline server/controllers/profile-controller.js]: Again please be more careful with documentation 🙂

File: server/controllers/profile-controller.js [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+    try {
+      return User.findAll()
+        .then(data => {
+          if (!data) {
+            errorResponse(res, 'No User(s)', 400);
+          }
+          successResponse(
+            res,
+            {
+              user: data
```
Comment (andela-pessien) [inline server/controllers/profile-controller.js:63]: Again no filtering of the user record - please correct

**Review by andela-pessien — CHANGES_REQUESTED** [review]
@mezlet Also please try to keep the commit histories of different branches you're working on separate.

File: server/controllers/profile-controller.js (hunk 1 above)
Comment (mezlet) [reply server/controllers/profile-controller.js]: ok I will work on this

File: server/controllers/profile-controller.js (hunk 2 above)
Comment (mezlet) [reply server/controllers/profile-controller.js:63]: please I don't understand what you mean by filtering user record

File: server/middlewares/index.js [hunk 3]
```diff
@@ -6,7 +6,7 @@ const { User } = db;
  *Users endpoint middlewares
  *
  * @export
- * @class UsersMiddleware
+ * @class UsersMiddlewares
  */
 export default class UsersMiddleware {
```
Comment (andela-pessien) [inline server/middlewares/index.js]: Please update the class name to match the new documentation

File: server/utils/helpers.js [hunk 4]
```diff
@@ -1,3 +1,4 @@
+import '@babel/polyfill';
```
Comment (andela-pessien) [inline server/utils/helpers.js]: This import is unnecessary

File: test/users.spec.js [hunk 5]
```diff
@@ -11,6 +11,13 @@ const user1 = {
   email: faker.internet.email(),
   password: faker.random.alphaNumeric(10)
 };
+
+const user2 = {
+  username: faker.random.alphaNumeric(10),
```
Comment (andela-pessien) [inline test/users.spec.js]: Consider the DRY principle in setting up your tests

**Review by andela-pessien — CHANGES_REQUESTED** [review]

File: test/users.spec.js [hunk 6]
```diff
@@ -32,6 +31,21 @@ describe('Authentication', () => {
         });
     });
 
+    it('should register a user with valid details', done => {
```
Comment (andela-pessien) [inline test/users.spec.js]: Please sort out the duplicate test (is it actually supposed to be a _test_ or is it setup for something?)

File: server/utils/helpers.js (hunk 4 above)
Comment (andela-pessien) [reply server/utils/helpers.js]: Move this to your entrypoint (`index.js`)

**Review by andela-pessien — CHANGES_REQUESTED** [review]

File: server/utils/helpers.js (hunk 4 above)
Comment (mezlet) [reply server/utils/helpers.js]: I have moved it

File: server/middlewares/index.js (hunk 3 above)
Comment (mezlet) [reply server/middlewares/index.js]: Acknowledged

File: test/users.spec.js (hunk 6 above)
Comment (mezlet) [reply test/users.spec.js]: Ok. I will look into this

File: server/middlewares/index.js (hunk 3 above)
Comment (mezlet) [reply server/middlewares/index.js]: I have fixed it

File: test/users.spec.js (hunk 5 above)
Comment (mezlet) [reply test/users.spec.js]: I have implemented the changes

File: test/users.spec.js (hunk 6 above)
Comment (mezlet) [reply test/users.spec.js]: sorted out

**Review by andela-pessien — APPROVED** [review]

**Pull request merged**